### PR TITLE
ci/linux: Upload separated debug symbols

### PR DESCRIPTION
**Pull request by lat9nq**
Creates a new archive with a debug suffix that contains the debug symbols from compiling yuzu for mainline. The yuzu executable also gets a GNU debug link to the symbols file. ci/linux: Compile with debug symbols and upload separately

Currently only uploads for yuzu but yuzu-cmd or other future executables can be added to the for-loop's parameters.

### CODE REVIEW

File: .ci/scripts/linux/upload.sh [hunk 1]
```diff
@@ -59,4 +59,9 @@ if [ "${RELEASE_NAME}" = "mainline" ] || [ "${RELEASE_NAME}" = "early-access" ];
     cp "build/${APPIMAGE_NAME}" "${DIR_NAME}/yuzu-${RELEASE_NAME}.AppImage"
 fi
 
+# Copy debug symbols to artifacts
+cd build/bin
+tar $COMPRESSION_FLAGS *.debug "${ARTIFACTS_DIR}/${REV_NAME}-debug.tar.xz"
```
Comment (liamwhite) [inline .ci/scripts/linux/upload.sh:64]: The archive name needs to go before the files you include (see CI failure)

File: .ci/scripts/linux/docker.sh [hunk 2]
```diff
@@ -31,6 +31,19 @@ ccache -s
...
 
+# Separate debug symbols from specified executables
+for EXE in yuzu; do
+    EXE_PATH="bin/$EXE"
+    # Copy debug symbols out
+    objcopy --only-keep-debug $EXE_PATH $EXE_PATH.debug
+    # Add debug link and strip debug symbols
+    objcopy -g --add-gnu-debuglink=$EXE_PATH.debug $EXE_PATH $EXE_PATH.out
+    # Overwrite original with stripped copy
+    cp $EXE_PATH.out $EXE_PATH
```
Comment (liamwhite) [inline .ci/scripts/linux/docker.sh:42]:
```suggestion
    mv $EXE_PATH.out $EXE_PATH
```

File: .ci/scripts/linux/docker.sh [hunk 3]
```diff
@@ -12,7 +12,7 @@ mkdir build || true && cd build
 cmake .. \
       -DBoost_USE_STATIC_LIBS=ON \
       -DCMAKE_BUILD_TYPE=Release \
-      -DCMAKE_CXX_FLAGS="-march=x86-64-v2" \
+      -DCMAKE_CXX_FLAGS="-march=x86-64-v2 -g" \
```
Comment (liamwhite) [inline .ci/scripts/linux/docker.sh:15]: Any specific reason not to use RelWithDebInfo, as opposed to passing -g?

Comment (lat9nq) [reply .ci/scripts/linux/docker.sh:15]: I was worried it'd be -O2. It isn't after looking, -O3 actually.

File: .ci/scripts/linux/docker.sh [hunk 4]
```diff
@@ -31,6 +31,19 @@ ccache -s
...
+    EXE_PATH="bin/$EXE"
+    # Copy debug symbols out
+    objcopy --only-keep-debug $EXE_PATH $EXE_PATH.debug
+    # Add debug link and strip debug symbols
+    objcopy -g --add-gnu-debuglink=$EXE_PATH.debug $EXE_PATH $EXE_PATH.out
+    # Overwrite original with stripped copy
+    mv $EXE_PATH.out $EXE_PATH
+done
+# Strip debug symbols from all executables
+find -type f bin/ -not -regex '.*.debug' -exec strip -g {} ';'
```
Comment (liamwhite) [inline .ci/scripts/linux/docker.sh:45]:
```suggestion
find bin/ -type f -not -regex '.*.debug' -exec strip -g {} ';'
```

**Review by liamwhite — APPROVED** [review]

**Pull request merged**